### DISCUSSION TITLE
Intel® 64 and IA-32 Architectures Software Developer’s Manual , which use  a assenbler ?   Microsoft   MASM assembler or Netwide  assembler(NASM)?

### DISCUSSION THREAD

**Discussion by hangshan-boop** [community/community#136933] [Programming Help]
### Select Topic Area

Question

### Body

This book that Intel® 64 and IA-32 Architectures Software Developer’s Manual , which use  a assenbler ?   Microsoft   MASM assembler or Netwide  assembler(NASM)?
![Screenshot_20240827_185545_cn wps moffice_eng](https://github.com/user-attachments/assets/6d633070-7b34-4f0f-a73e-d73124159ac3)
I need to help

**Comment by LiteBrite82** [comment]
Thanks for posting in the GitHub Community, @hangshan-boop ! 

We’ve moved your post to our Programming Help 🧑‍💻 category, which is more appropriate for this type of discussion.

Please review our [guidelines](https://github.com/orgs/community/discussions/51384) about the Programming Help category for more information.

**Comment by tpmst** [comment]
The Intel® 64 and IA-32 Architectures Software Developer’s Manual you're referring to is a comprehensive source that details the functionality of the Intel 64 and IA-32 architectures. It is designed to be useful for programmers who are using any assembler, such as Microsoft MASM or NASM. The manual itself is not specific to any assembler but provides the necessary information to use any such tools effectively.

## Assembler Usage with Intel Manuals:
### MASM (Microsoft Macro Assembler):

1. MASM provides a Windows-centric assembly language development environment and is tightly integrated with Microsoft development tools.
2. MASM might be preferable if you are developing applications in a Microsoft Windows environment due to its native integration.

### NASM (Netwide Assembler):

1. NASM is widely used for its simplicity and ability to compile for both Windows and UNIX-like (Linux, macOS) operating systems.
2. It supports a wide range of object file formats and works well for cross-platform development.

## Conclusion

The examples and syntactical conventions in the Intel manuals can generally be adapted for use with either assembler. However, you might need to adjust syntax or conventions slightly depending on the assembler’s specific requirements. For direct compatibility or examples, it’s often useful to check the documentation or community forums specific to MASM or NASM for guidance on implementing Intel's assembly instructions.

## Additions

Sorry, but your question wasn't specific enough for me to provide the exact information you were likely looking for. I hope my response was still helpful

**Comment by hangshan-boop** [comment]
I want to know&nbsp; instructions in manual that need to use which assembler! Thank you




---Original---
From: ***@***.***&gt;
Date: Wed, Sep 11, 2024 13:51 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [community/community] Intel® 64 and IA-32 Architectures Software Developer’s Manual , which use  a assenbler ?   Microsoft   MASM assembler or Netwide  assembler(NASM)? (Discussion #136933)




 
The Intel® 64 and IA-32 Architectures Software Developer’s Manual you're referring to is a comprehensive source that details the functionality of the Intel 64 and IA-32 architectures. It is designed to be useful for programmers who are using any assembler, such as Microsoft MASM or NASM. The manual itself is not specific to any assembler but provides the necessary information to use any such tools effectively.
 
Assembler Usage with Intel Manuals:
 
MASM (Microsoft Macro Assembler):
  
MASM provides a Windows-centric assembly language development environment and is tightly integrated with Microsoft development tools.
 
MASM might be preferable if you are developing applications in a Microsoft Windows environment due to its native integration.
  
NASM (Netwide Assembler):
  
NASM is widely used for its simplicity and ability to compile for both Windows and UNIX-like (Linux, macOS) operating systems.
 
It supports a wide range of object file formats and works well for cross-platform development.
  
Conclusion
 
The examples and syntactical conventions in the Intel manuals can generally be adapted for use with either assembler. However, you might need to adjust syntax or conventions slightly depending on the assembler’s specific requirements. For direct compatibility or examples, it’s often useful to check the documentation or community forums specific to MASM or NASM for guidance on implementing Intel's assembly instructions.
 
Additions
 
Sorry, but your question wasn't specific enough for me to provide the exact information you were likely looking for. I hope my response was still helpful

**Reply by tpmst** [reply]
This manual provides comprehensive information about the instruction set architecture, including detailed descriptions of instructions used in Intel processors, but it does not specify a particular assembler.

## Assembler
The choice of assembler often depends on your development environment, the operating system you are targeting, and personal preference.
So you need to choose your self.

**Comment by hangshan-boop** [comment]
I want to know if Intel hold their assembler.&nbsp; Before itself hold assembler



---Original---
From: ***@***.***&gt;
Date: Wed, Sep 11, 2024 14:27 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [community/community] Intel® 64 and IA-32 Architectures Software Developer’s Manual , which use  a assenbler ?   Microsoft   MASM assembler or Netwide  assembler(NASM)? (Discussion #136933)




 
This manual provides comprehensive information about the instruction set architecture, including detailed descriptions of instructions used in Intel processors, but it does not specify a particular assembler.
 
Assembler
 
The choice of assembler often depends on your development environment, the operating system you are targeting, and personal preference.
 So you need to choose your self.

**Reply by tpmst** [reply]
Could you go more in-depth with your problem and describe the problem better.

**Comment by hangshan-boop** [comment]
Intel hold C++ complier and Fortran compiler .&nbsp; They are added to Visual Studio&nbsp; in&nbsp; Microsoft



---Original---
From: ***@***.***&gt;
Date: Wed, Sep 11, 2024 14:27 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [community/community] Intel® 64 and IA-32 Architectures Software Developer’s Manual , which use  a assenbler ?   Microsoft   MASM assembler or Netwide  assembler(NASM)? (Discussion #136933)




 
This manual provides comprehensive information about the instruction set architecture, including detailed descriptions of instructions used in Intel processors, but it does not specify a particular assembler.
 
Assembler
 
The choice of assembler often depends on your development environment, the operating system you are targeting, and personal preference.
 So you need to choose your self.

**Comment by hangshan-boop** [comment]
Intel hold itself assembler?



---Original---
From: ***@***.***&gt;
Date: Wed, Sep 11, 2024 16:07 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [community/community] Intel® 64 and IA-32 Architectures Software Developer’s Manual , which use  a assenbler ?   Microsoft   MASM assembler or Netwide  assembler(NASM)? (Discussion #136933)




 
Could you go more in-depth with your problem and describe the problem better.

**Comment by hangshan-boop** [comment]
Do intel hold itself assembler?



---Original---
From: ***@***.***&gt;
Date: Wed, Sep 11, 2024 16:07 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [community/community] Intel® 64 and IA-32 Architectures Software Developer’s Manual , which use  a assenbler ?   Microsoft   MASM assembler or Netwide  assembler(NASM)? (Discussion #136933)




 
Could you go more in-depth with your problem and describe the problem better.

**Reply by tpmst** [reply]
Yes, Intel historically developed its own assembler, often referred to as the Intel Assembler or IASM. This assembler was designed to work specifically with Intel's CPUs, supporting the instruction sets and architectures developed by Intel.

**Comment by hangshan-boop** [comment]
Now ,who do replaced it?



---Original---
From: ***@***.***&gt;
Date: Wed, Sep 11, 2024 16:17 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [community/community] Intel® 64 and IA-32 Architectures Software Developer’s Manual , which use  a assenbler ?   Microsoft   MASM assembler or Netwide  assembler(NASM)? (Discussion #136933)




 
Yes, Intel historically developed its own assembler, often referred to as the Intel Assembler or IASM. This assembler was designed to work specifically with Intel's CPUs, supporting the instruction sets and architectures developed by Intel.

**Reply by tpmst** [reply]
Do you want to know which assembler replaced the ISMA assembler?

**Comment by hangshan-boop** [comment]
Yes



---Original---
From: ***@***.***&gt;
Date: Wed, Sep 11, 2024 16:47 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [community/community] Intel® 64 and IA-32 Architectures Software Developer’s Manual , which use  a assenbler ?   Microsoft   MASM assembler or Netwide  assembler(NASM)? (Discussion #136933)




 
Do you want to know which assembler replaced the ISMA assembler?

**Comment by tpmst** [comment]
MASM (Microsoft Macro Assembler) is often considered the de facto replacement for Intel’s assembler in many Windows-based environments. MASM has been widely used for its integration with Microsoft's development tools and for its support of advanced Windows features.

Additionally, for environments outside of Windows or for developers seeking cross-platform solutions, NASM (Netwide Assembler) serves as another excellent alternative. NASM is known for its simplicity and effectiveness and is widely respected in the open-source community.

While MASM and NASM are not direct products or releases by Intel, they are the practical successors for developers looking to write assembly code optimized for Intel processors, especially in the absence of a modern dedicated Intel assembler. These assemblers have effectively filled the gap left by Intel’s own assembler and have become the standard tools for assembly language programming on Intel architecture.

If this was the answer to your question please mark it as answered.

**Comment by hangshan-boop** [comment]
MASM is support&nbsp;Intel® 64 and IA-32 Architectures Software Developer’s Manual ,is ture or error?



---Original---
From: ***@***.***&gt;
Date: Wed, Sep 11, 2024 17:02 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [community/community] Intel® 64 and IA-32 Architectures Software Developer’s Manual , which use  a assenbler ?   Microsoft   MASM assembler or Netwide  assembler(NASM)? (Discussion #136933)




 
MASM (Microsoft Macro Assembler) is often considered the de facto replacement for Intel’s assembler in many Windows-based environments. MASM has been widely used for its integration with Microsoft's development tools and for its support of advanced Windows features.

**Comment by tpmst** [comment]
Yes, that's true. MASM (Microsoft Macro Assembler) supports the Intel® 64 and IA-32 architectures and is fully compatible with the instruction sets described in the Intel® 64 and IA-32 Architectures Software Developer’s Manual. But it only is true for Windows-based environments.

**Comment by hangshan-boop** [comment]
Do you have materials that prove MASM that useIntel® 64 and IA-32 Architectures Software Developer’s Manual ?



---Original---
From: ***@***.***&gt;
Date: Wed, Sep 11, 2024 17:54 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [community/community] Intel® 64 and IA-32 Architectures Software Developer’s Manual , which use  a assenbler ?   Microsoft   MASM assembler or Netwide  assembler(NASM)? (Discussion #136933)




 
Yes, that's true. MASM (Microsoft Macro Assembler) supports the Intel® 64 and IA-32 architectures and is fully compatible with the instruction sets described in the Intel® 64 and IA-32 Architectures Software Developer’s Manual. But it only is true for Windows-based environments.

**Comment by hangshan-boop** [comment]
Thank you!!!



---Original---
From: ***@***.***&gt;
Date: Wed, Sep 11, 2024 17:54 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [community/community] Intel® 64 and IA-32 Architectures Software Developer’s Manual , which use  a assenbler ?   Microsoft   MASM assembler or Netwide  assembler(NASM)? (Discussion #136933)




 
Yes, that's true. MASM (Microsoft Macro Assembler) supports the Intel® 64 and IA-32 architectures and is fully compatible with the instruction sets described in the Intel® 64 and IA-32 Architectures Software Developer’s Manual. But it only is true for Windows-based environments.

**Comment by tpmst** [comment]
Microsoft’s official documentation for MASM provides insights into how MASM is used to write applications for Windows, which runs on Intel’s IA-32 and Intel 64 architectures.

MASM Documentation: Check Microsoft's official documentation or MSDN (Microsoft Developer Network) for details on MASM. While explicit references linking MASM to Intel’s manual might not be abundant, the compatibility is implicit given that MASM is designed to work with Intel architectures.

https://learn.microsoft.com/en-us/cpp/assembler/masm/microsoft-macro-assembler-reference?view=msvc-170

If this was the answer to your question please mark it as answered.

**Comment by hangshan-boop** [comment]
Thank you!
I find answer.



---Original---
From: ***@***.***&gt;
Date: Wed, Sep 11, 2024 19:54 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [community/community] Intel® 64 and IA-32 Architectures Software Developer’s Manual , which use  a assenbler ?   Microsoft   MASM assembler or Netwide  assembler(NASM)? (Discussion #136933)




 
Microsoft’s official documentation for MASM provides insights into how MASM is used to write applications for Windows, which runs on Intel’s IA-32 and Intel 64 architectures.
 
MASM Documentation: Check Microsoft's official documentation or MSDN (Microsoft Developer Network) for details on MASM. While explicit references linking MASM to Intel’s manual might not be abundant, the compatibility is implicit given that MASM is designed to work with Intel architectures.
 
https://learn.microsoft.com/en-us/cpp/assembler/masm/microsoft-macro-assembler-reference?view=msvc-170

**Comment by tpmst** [comment]
Ok, then mark this discussion as answered and I hope you enjoy your new knowledge.

**Comment by hangshan-boop** [comment]
I want to know that IDA Pro support Intel syntax specifically.



---Original---
From: ***@***.***&gt;
Date: Wed, Sep 11, 2024 19:54 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [community/community] Intel® 64 and IA-32 Architectures Software Developer’s Manual , which use  a assenbler ?   Microsoft   MASM assembler or Netwide  assembler(NASM)? (Discussion #136933)




 
Microsoft’s official documentation for MASM provides insights into how MASM is used to write applications for Windows, which runs on Intel’s IA-32 and Intel 64 architectures.
 
MASM Documentation: Check Microsoft's official documentation or MSDN (Microsoft Developer Network) for details on MASM. While explicit references linking MASM to Intel’s manual might not be abundant, the compatibility is implicit given that MASM is designed to work with Intel architectures.
 
https://learn.microsoft.com/en-us/cpp/assembler/masm/microsoft-macro-assembler-reference?view=msvc-170

**Comment by hangshan-boop** [comment]
How to do mark&nbsp; the answer



---Original---
From: ***@***.***&gt;
Date: Wed, Sep 11, 2024 20:47 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [community/community] Intel® 64 and IA-32 Architectures Software Developer’s Manual , which use  a assenbler ?   Microsoft   MASM assembler or Netwide  assembler(NASM)? (Discussion #136933)




 
Ok, then mark this discussion as answered and I hope you enjoy your new knowledge.

**Comment by tpmst** [comment]
Yes, IDA Pro does support Intel syntax for disassembly. It supports a wide range of architectures and includes the capability to switch between different assembly syntaxes, including Intel and AT&T syntax for x86 architecture.

Below here should be a button for choosing it as a answer.

**Comment by hangshan-boop** [comment]
I want to know that it support Intel® 64 and IA-32 Architectures Software Developer’s Manual or AMD manual ,either&nbsp; MASM syntax.Thank you!!



---Original---
From: ***@***.***&gt;
Date: Wed, Sep 11, 2024 20:54 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [community/community] Intel® 64 and IA-32 Architectures Software Developer’s Manual , which use  a assenbler ?   Microsoft   MASM assembler or Netwide  assembler(NASM)? (Discussion #136933)




 
Yes, IDA Pro does support Intel syntax for disassembly. It supports a wide range of architectures and includes the capability to switch between different assembly syntaxes, including Intel and AT&amp;T syntax for x86 architecture.
 
Below here should be a button for choosing it as a answer.

**Comment by tpmst** [comment]
No it only supports Intel 64 Architectures and not AMD.

**Comment by hangshan-boop** [comment]
It&nbsp; support&nbsp;Intel® 64 and IA-32 Architectures Software Developer’s Manual ,is true?&nbsp; can&nbsp; you sent a associated article to me?



---Original---
From: ***@***.***&gt;
Date: Wed, Sep 11, 2024 21:15 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [community/community] Intel® 64 and IA-32 Architectures Software Developer’s Manual , which use  a assenbler ?   Microsoft   MASM assembler or Netwide  assembler(NASM)? (Discussion #136933)




 
No it only supports Intel 64 Architectures and not AMD.

**Comment by tpmst** [comment]
I wont answer any further questions for something like this maby you could google it yourself.

**Comment by hangshan-boop** [comment]
In Chinese , Goole isn't forbidden. We don't find English article.&nbsp; I met a problem that I read the art of computer programming that I don' how to write a stack,function and array. Because assembler use intel syntax have lot of assembler.



---Original---
From: ***@***.***&gt;
Date: Wed, Sep 11, 2024 21:23 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [community/community] Intel® 64 and IA-32 Architectures Software Developer’s Manual , which use  a assenbler ?   Microsoft   MASM assembler or Netwide  assembler(NASM)? (Discussion #136933)




 
I wont answer any further questions for something like this maby you could google it yourself.

**Comment by hangshan-boop** [comment]
God you are rest



---Original---
From: ***@***.***&gt;
Date: Wed, Sep 11, 2024 21:23 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [community/community] Intel® 64 and IA-32 Architectures Software Developer’s Manual , which use  a assenbler ?   Microsoft   MASM assembler or Netwide  assembler(NASM)? (Discussion #136933)




 
I wont answer any further questions for something like this maby you could google it yourself.
